wallet/src/main/java/com/android/identity_credential/wallet/credman/CredmanPresentationActivity.kt

```diff
@@ -306,34 +241,43 @@ class CredmanPresentationActivity : FragmentActivity() {
                         Crypto.digest(Algorithm.SHA256, clientID.toByteArray())
                     )
                 }
-                // Create ISO DeviceResponse
-                createMDocDeviceResponse(
-                    credentialId,
-                    dataElements,
-                    encodedSessionTranscript,
-                    onComplete = { deviceResponse, credential ->
-                        credential.increaseUsageCount()
-                        // Create the openid4vp response
-                        val responseJson = JSONObject()
-                        responseJson.put(
-                            "vp_token",
-                            Base64.encodeToString(
-                                deviceResponse,
-                                Base64.NO_WRAP or Base64.URL_SAFE
-                            )
+                lifecycleScope.launch {
+                    val mdocCredential = getMdocCredentialForCredentialId(credentialId)
+                    val documentCborBytes = showPresentmentFlow(
+                        activity = this@CredmanPresentationActivity,
+                        walletApp = walletApp,
+                        documentRequest = DocumentRequest(dataElements),
+                        mdocCredential = mdocCredential,
+                        trustPoint = null,
+                        encodedSessionTranscript = encodedSessionTranscript,
+                        showConsentPrompt = false
+                    )
+                    // Create ISO DeviceResponse
+                    val deviceResponseGenerator =
+                        DeviceResponseGenerator(Constants.DEVICE_RESPONSE_STATUS_OK)
+                    deviceResponseGenerator.addDocument(documentCborBytes)
+                    val deviceResponse = deviceResponseGenerator.generate()
```

Comment:
Do you think we could factor out the call to showPresentmentFlow() + device response generation to a helper function so we only have a single callsite?

Reply:
the generated response jsons are almost the same, one adds a `token` vs `vp_token`, I'll add a boolean for **openid4vp** branch in the function